### ISSUE TITLE
License key app must be the first installed of the EE bundle

### ISSUE THREAD

**Issue by davitol** [owncloud/market#165]
Installing EE app bundle:

![screen shot 2017-09-14 at 17 13 33](https://user-images.githubusercontent.com/10596560/30437892-135d69d8-9970-11e7-9eea-344b689f3a37.png)

**Comment by patrickjahns** [comment]
Can you please verify that there is no `'license-key'` set in you config.php? 

It doesn't show the "start enterprise trail button" which is usually only hidden if you have a license key defined

**Comment by patrickjahns** [comment]
I can not reproduce - when no license_key is set in `config.php` I am not able to download the enterprise applications. 

When I define an enterprise_key I don't see these errors. Can you please provide exact steps to reproduce

**Comment by davitol** [comment]
1. Install ownCloud from owncloud-daily-stable10.tar.bz2
2. I checked that in config.php file it was set ```'license-key' => 'demo-20321531-XXXXXXXXXXXXX-XXXX',```
3. Go to market app
4. Add Personal API key
5. Install EE App bundle.

**Comment by davitol** [comment]
additonal info;
oc_appconfig table does not contain any record for enterprise license key, only these rows related to enterprise_key:


|enterprise_key | enabled | yes |
| enterprise_key | installed_version | 0.1.3 |
| enterprise_key | signed | true |
| enterprise_key | types | filesystem

**Comment by patrickjahns** [comment]
I could reproduce this.

The problem lies within the frontend handling. It sends apparently individual requests:

![image](https://user-images.githubusercontent.com/6403243/30476812-78eda4de-9a0b-11e7-96e0-f9be8411af60.png)

**Comment by patrickjahns** [comment]
Talked with @felixheidecke since he has provided the frontend and also created the frontend flow for the "start enterprise trail" button.

The short term solution would be, that the frontend checks if it's the enterprise bundle it will try to install and then try to install the license key first.

For the longterm we need to define a proper solution - ideas could be:

1) ship the enterprise license key app by default (if we don't allow apps being dependent on other apps that are not in core)

2) extend the marketapp with a "install bundle" route which has an internal logic for the order of installing applications?

3) adding to 2) - have the marketplace api return "installInfo" for bundles. It could look like something:
```
[
    {
        "id": "bundle",
        ...
        "products": [
            {
                "id": "app1", 
                ....
            },
            {
                "id": "app2", 
                .....
            },
            {
                "id": "app3", 
                .....
            },
        ]
     "installInfo" {
        order: {
           "1": "app2"
           "2": "app3"
           "3": "app1"
         } 
      }
    }
]
```

We need to define the behavior here

**Comment by PVince81** [comment]
Let's not rush this. Currently no one is able to reach this code path on the actual marketplace.

Trial button works fine, so let's defer the fix to post-10.0.3 but before marketplace is ready for this.

**Comment by patrickjahns** [comment]
#168 will address the issue specific now for the enterprise bundle. It works - but the frontend is controlling the behavior. 

@DeepDiver1975 
Maybe you can input how we should handle this properly
